src/Pos/Explorer/Web/Server.hs

```diff
@@ -532,9 +554,23 @@ unwrapOrThrow = either (throwM . Internal) pure
 
 -- | Get transaction from memory (STM) or throw exception.
 fetchTxFromMempoolOrFail :: ExplorerMode m => TxId -> m TxAux
-fetchTxFromMempoolOrFail txId =
-    view (mpLocalTxs . at txId) <$> getMemPool >>=
-    maybeThrow (Internal "Transaction not found in the mempool")
+fetchTxFromMempoolOrFail txId = do
+    memPoolTxs        <- localMemPoolTxs
+    let memPoolTxsSize = HM.size memPoolTxs
+
+    logDebug $ sformat ("Mempool size "%int%" found!") memPoolTxsSize
```

Comment:
@sectore My experience says that readability is often related to the dreaded LOC. And honestly, I don't see the point of "saving" one line of code - what's the point of that? If we are saving the LOC so we can have less maintenance - _then we can have a single line of code that contains all logic_. But that's not going to help us maintain the code easier, is it? Because the code maintainability seems to be linked to code readability.
Adding a single "variable" before helps the person understand the context - the variable name explains what it does and it's contents show that is does exactly that.
If you have to add 1000 lines of code but you make the code more readable, I'm for it.

I do agree that readability is variable, and there are some places where I could have just gone "that's good enough for me", but then rewrote it so it's more verbose so an idiot (like me) can understand it if he sees it for the first time.

That should be the readability standard. Idiot passes required for understanding (IPRU) :smile: 

https://serokell.slack.com/archives/G54P1SQCS/p1499928889757999